core/src/main/java/com/oath/oak/MemoryManager.java

```diff
@@ -8,32 +8,30 @@
 
 
 
+import io.netty.buffer.ByteBuf;
+
 import java.nio.ByteBuffer;
 
 
 public class MemoryManager {
     private final OakMemoryAllocator keysMemoryAllocator;
-    private final OakMemoryAllocator valuesMemoryAllocator;
-
-    public MemoryManager(OakMemoryAllocator valuesMemoryAllocator, OakMemoryAllocator keysMemoryAllocator) {
-        assert valuesMemoryAllocator != null;
-        assert keysMemoryAllocator != null;
+    private final NettyMemoryAllocator valuesMemoryAllocator;
 
-        this.valuesMemoryAllocator = valuesMemoryAllocator;
+    public MemoryManager(OakMemoryAllocator keysMemoryAllocator) {
+        this.valuesMemoryAllocator = new NettyMemoryAllocator();
         this.keysMemoryAllocator = keysMemoryAllocator;
     }
 
-    public ByteBuffer allocate(int size) {
+    public ByteBuf allocate(int size) {
         return valuesMemoryAllocator.allocate(size);
     }
 
     public void close() {
-        valuesMemoryAllocator.close();
         keysMemoryAllocator.close();
     }
 
-    void release(ByteBuffer bb) {
-        valuesMemoryAllocator.free(bb);
+    void release(ByteBuf bb) {
+        bb.release();
```

Comment:
First, the allocation is done via allocator while release/free not. Second, the release is conducted under reference counting, and it looks like we do not increase the reference count under each access, so again I wonder if it is the correct usage...

boolean release()
Decreases the reference count by 1 and deallocates this object if the reference count reaches at 0.
Returns:
true if and only if the reference count became 0 and this object has been deallocated

https://netty.io/4.0/api/io/netty/util/ReferenceCounted.html

Reply:
Yes this is the way to use it. when the refcount gets to 0 the bytebuf is automatically released back to the pool. Its the user responsibility to decrease the refcount using the release function.
I dont think we currently need to increase refcount on each access because we have the handle lock.